### PR TITLE
New blog about Thanos adoption at Aiven

### PR DESCRIPTION
**Pull request by jkowall**
This is a new blog I have written with two other co-authors here at Aiven on our adoption of Thanos at Aiven. I think it's a great contribution, and we hope to keep doing so upstream and pushing Thanos forward in the community. I would appreciate any reviews on the blog. 

Thanks!

### CODE REVIEW

File: docs/blog/2023-06-08-thanos-at-aiven.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+We are also paying roughly 25% for the storage costs. M3DB has a total of 54TB of storage provisioned today, at a cost of $4320 per month. We could house 216TB of storage for the same cost with Thanos. We are currently generating about 750GB per day, which means we can keep almost a year of metrics for the same cost as M3DB. Additionally, we are backing up M3DB which is using 33TB of object storage at a cost of $1,320 per month. With object storage, we have the added cost for the networking, this is around $1800 per month in additional costs. Here are our estimated costs:
+
+* M3: $38480
+
+* Thanos without any historical data in storage: $16,939
+* Thanos with one month retention in object storage: $17,368
+* Thanos with six months retention: $19,703
+* Thanos with a years retention: $22,447
+* Thanos with 2 years retention: $27,955
+* Thanos with 3 years retention: $33,423
```
Comment (bwplotka) [inline docs/blog/2023-06-08-thanos-at-aiven.md:96]: Hm, Is your obj storage cost alone ~18k USD for 3y data? Sounds bit off and expensive 🤔

File: docs/blog/2023-06-08-thanos-at-aiven.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+thanosreceiver | n2d-highcpu-96 | $1,924 | 6 | $11,544
+thanosquery | n2d-highcpu-16 | $321.57 | 6 | $1,929
+thanosstore | n2d-highmem-8 | $293.90 | 3 | $881.7
+thanoscompactor | n2d-highmem-8 | $293.90 | 1 | $293.90
+redis | n2d-highmem-4 | $147.45 | 2 | $294.9
+THANOS TOTAL | | $14,942.68
+
+We are also paying roughly 25% for the storage costs. M3DB has a total of 54TB of storage provisioned today, at a cost of $4320 per month. We could house 216TB of storage for the same cost with Thanos. We are currently generating about 750GB per day, which means we can keep almost a year of metrics for the same cost as M3DB. Additionally, we are backing up M3DB which is using 33TB of object storage at a cost of $1,320 per month. With object storage, we have the added cost for the networking, this is around $1800 per month in additional costs. Here are our estimated costs:
+
+* M3: $38480
```
Comment (bwplotka) [inline docs/blog/2023-06-08-thanos-at-aiven.md:89]: For what retention?

File: docs/blog/2023-06-08-thanos-at-aiven.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+* Thanos without any historical data in storage: $16,939
+* Thanos with one month retention in object storage: $17,368
+* Thanos with six months retention: $19,703
+* Thanos with a years retention: $22,447
+* Thanos with 2 years retention: $27,955
+* Thanos with 3 years retention: $33,423
+
+As you can see, the cost savings are significant here. [Alexander Rickardsson](https://github.com/alxric) implemented AZ awareness in Thanos upstream now, which reduced our replication factor from 3 to 2 on Google Cloud Platform. https://github.com/thanos-io/thanos/pull/6369#event-9357572391
```
Comment (bwplotka) [inline docs/blog/2023-06-08-thanos-at-aiven.md:98]:
```suggestion
As you can see, the cost savings are significant here. Plus, there is ongoing work for further cost optimizations in Thanos. For example, [Alexander Rickardsson](https://github.com/alxric) implemented AZ awareness in Thanos upstream now, which reduced our replication factor from 3 to 2 on Google Cloud Platform. https://github.com/thanos-io/thanos/pull/6369#event-9357572391 
```

File: docs/blog/2023-06-08-thanos-at-aiven.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+* Thanos with six months retention: $19,703
+* Thanos with a years retention: $22,447
+* Thanos with 2 years retention: $27,955
+* Thanos with 3 years retention: $33,423
+
+As you can see, the cost savings are significant here. [Alexander Rickardsson](https://github.com/alxric) implemented AZ awareness in Thanos upstream now, which reduced our replication factor from 3 to 2 on Google Cloud Platform. https://github.com/thanos-io/thanos/pull/6369#event-9357572391 
+
+## Performance Gains
+
+The performance normally was much better in our ongoing reporting and alerting needs as well. Today we are using vmalert to drive our alerting pipeline, since M3DB is limited, there is no such thing as an alertmanager integration. This brings me to another issue we found with vmalert it would sometimes execute rules twice within the same group evaluation period. This, by default, would realign the result timestamp with the group evaluation start time, which would lead to failed and rejected writes. The timestamp issue was caused by samples with same timestamp but different value ), this was fixed by disabling this query time alignment ( datasource.queryTimeAlignment ).
```
Comment (bwplotka) [inline docs/blog/2023-06-08-thanos-at-aiven.md:102]: Why vmalert? Any specific reason why not Thanos Ruler?

**Review by bwplotka — DISMISSED** [review]
Epic! Thanks for this. Some high level comments, but overall looks amazing 💪🏽

File: docs/blog/2023-06-08-thanos-at-aiven.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+With these options, we had to discount Mimir even though it’s great technology, it’s controlled by a single company and restrictively licensed. Similar to Elasticsearch, we would rather not have to go down that path in the future. This meant the main options we had were Cortex and Thanos. Cortex has similar problems to M3DB. There are only a couple of companies using Cortex, since Mimir is an improved fork, and those that aren't concerned about the fact that the license could change next week have already switched over. The Mimir community is controlled by a single company, which is not ideal. Aligning with software foundations is the right way to protect a project and ensure a vibrant and diverse community. Thanos was our target. We would start by seeing if we can use it internally and then determine if we could build a product.
+
+We began the process with creating a plan on how we would test the solution to see how it met our needs. This would be a Proof of Concept (POC) to see if it were feasible. Here was our basic setup today
+
+  ```mermaid
```
Comment (saswatamcode) [inline docs/blog/2023-06-08-thanos-at-aiven.md:31]: This is an awesome diagram but doesn't work in our Hugo theme for the website. I wonder if we can link a static image here for the time being, and comment the mermaid one, until we have support for it. 🙂

**Review by saswatamcode — DISMISSED** [review]
Awesome! 🚀 
LGTM mod small nit about the diagram and @bwplotka's comments.

File: docs/blog/2023-06-08-thanos-at-aiven.md (hunk 1 above)
Comment (jkowall) [reply docs/blog/2023-06-08-thanos-at-aiven.md:96]: We have a lot of metrics and manage a large fleet. We also use Thanos (m3) for our billing data which is quite detailed at the service level. This is really the costs, but look how it compares to the cost of M3.

File: docs/blog/2023-06-08-thanos-at-aiven.md (hunk 2 above)
Comment (jkowall) [reply docs/blog/2023-06-08-thanos-at-aiven.md:89]: I will add details on that. Thanks.

File: docs/blog/2023-06-08-thanos-at-aiven.md (hunk 4 above)
Comment (jkowall) [reply docs/blog/2023-06-08-thanos-at-aiven.md:102]: M3DB doesn't have a similar capability, so we are using vmalert. It was easier than switching everything over. We are happy with vmalert right now even though it's pull vs push.

File: docs/blog/2023-06-08-thanos-at-aiven.md (hunk 5 above)
Comment (jkowall) [reply docs/blog/2023-06-08-thanos-at-aiven.md:31]: Ahhh, too bad. I was worried about that. I will add images and leave the code in a comment?

**Comment by jkowall** [comment]
Not sure why the docs is giving an error on outstanding commits, can someone assist on it?  (@saswatamcode)

**Comment by saswatamcode** [comment]
Yes, @jkowall! Could you run `make docs` and commit? This formats the markdown file (tables/spacing/whitespaces etc). 🙂

File: docs/blog/2023-06-08-thanos-at-aiven.md (hunk 5 above)
Comment (saswatamcode) [reply docs/blog/2023-06-08-thanos-at-aiven.md]: Thanks!

**Comment by jkowall** [comment]
Looks good now, thanks @saswatamcode !

**Comment by jkowall** [comment]
Still can't merge it due to a CI failure, any advice would be helpful.

**Review by fpetkovski — APPROVED** [review]
Thanks for the write up 👍 I kicked the CI a couple of times since failures were flakes in e2e tests.

**Review by saswatamcode — APPROVED** [review]
Thanks for the epic blog!

**Pull request merged**